examples/gno.land/r/gnoland/valopers/valopers.gno

```diff
@@ -5,35 +5,154 @@ package valopers
 import (
 	"std"
 
-	"gno.land/p/demo/ownable"
-	govdao "gno.land/r/gov/dao"
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
 )
 
-// Valoper represents a validator operator profile.
+const daoPkgPath = "gno.land/r/gov/dao"
+
+const (
+	errValoperExists        = "valoper already exists"
+	errValoperMissing       = "valoper does not exist"
+	errInvalidAddressUpdate = "valoper updated address exists"
+
+	errNotGovDAO = "caller not govdao executor"
+)
+
+// valopers keeps track of all the active validator operators
+var valopers *avl.Tree // Address -> Valoper
+
+// Valoper represents a validator operator profile
 type Valoper struct {
-	ownable.Ownable // Embedding the Ownable type for ownership management.
+	Name        string // the display name of the valoper
+	Description string // the description of the valoper
 
-	DisplayName   string      // The display name of the valoper.
-	ValidatorAddr std.Address // The address of the validator.
-	// TODO: Add other valoper metadata as needed.
+	Address      std.Address // The bech32 gno address of the validator
+	PubKey       string      // the bech32 public key of the validator
+	P2PAddresses []string    // the publicly reachable P2P addresses of the validator
 }
 
-// Register registers a new valoper.
-// TODO: Define the parameters and implement the function.
-func Register( /* TBD */ ) {
-	panic("not implemented")
+// Register registers a new valoper
+func Register(v Valoper) {
+	assertGovDAOCaller()
```

Comment:
This realm is intended to be permissionless, allowing anyone to register themselves as validator operators. This process is similar to the `gaiad tx staking create-validator` command (https://hub.cosmos.network/main/validators/validator-setup.html#create-your-validator).

Once registered, the GovDAO can create a proposal to add the validator to the active valset. This is what I refer to as a "templated proposal".

Reply:
Understood, I thought we needed to have govdao safeguards

I dropped the govdao assertions:
[f04c482](https://github.com/gnolang/gno/pull/2509/commits/f04c482179c92256710b95e21f78285a2c7a4521)